src/components/FindMovie/FindMovie.tsx

```
@@ -1,56 +1,126 @@
+import { ResponseError } from '../../types/ReponseError';
+
+interface Props {
+  onAdd: (movie: Movie) => void;
+}
+
+const POSTER_FALLBACK = 'https://via.placeholder.com/300x445?text=No+Poster';
+
+function normalizeMovie(d: MovieData) {
+  return {
+    title: d.Title,
+    description: d.Plot,
+    imgUrl: d.Poster && d.Poster !== 'N/A' ? d.Poster : POSTER_FALLBACK,
+    imdbUrl: `https://www.imdb.com/title/${d.imdbID}`,
+    imdbId: d.imdbID,
+  };
+}
+
+function isError(x: MovieData | ResponseError): x is ResponseError {
+  return (x as ResponseError).Response === 'False';
+}
+
+export const FindMovie: React.FC<Props> = ({ onAdd }) => {
+  const [errorMessage, setErrorMessage] = useState('');
+  const [loading, setLoading] = useState(false);
+  const [preview, setPreview] = useState<Movie | null>(null);
+  const [query, setQuery] = useState('');
+
+  const onSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+    setLoading(true);
+
+    getMovie(query)
+      .then(data => {
+        if (isError(data)) {
+          setErrorMessage(data.Error || 'Movie not found');
+          setPreview(null);
+
+          return;
+        } else {
+          setPreview(normalizeMovie(data));
+        }
+      })
+      .catch(error => setErrorMessage(String(error)))
+      .finally(() => setLoading(false));
+    setQuery('');
+  };
+
+  const handleAdd = () => {
+    if (!preview) {
+      return;
+    }
+
+    onAdd(preview);
+
+    setErrorMessage('');
+    setLoading(false);
+    setQuery('');
+  };
 
-export const FindMovie: React.FC = () => {
   return (
     <>
-      <form className="find-movie">
+      <form className="find-movie" onSubmit={event => onSubmit(event)}>
         <div className="field">
           <label className="label" htmlFor="movie-title">
             Movie title
           </label>
 
           <div className="control">
             <input
+              value={query}
+              onChange={event => setQuery(event.target.value)}
```

The requirement is to hide the error message when the user starts typing a new title. You can achieve this by also clearing the `errorMessage` state in this `onChange` handler.